### PR TITLE
[feat]: [PL-27119]: API KEY Auth enablement

### PR DESCRIPTION
**Pull request by boopesh007**
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36222)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36222)

**Comment by boopesh007** [comment]
trigger messagecheck

**Review by DeepakPatankar — COMMENTED** [review]
> Reviewed 8 of 17 files at r1, 1 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36222)* status: 18 file(s) to review. 0 of 2 approvals obtained.

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36222)* status: 18 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/resources/ServiceResource.java` line 88 at r2](https://reviewable.io/reviews/harness/harness-core/36222#-N9fmf0c-lglcT-GkgPU:-N9fmf0c-lglcT-GkgPV:bnlrqaf) ([raw file](https://github.com/harness/harness-core/blob/fe5998ed83b0bac1fa189ad8ce6d64dfc0de1fb3/400-rest/src/main/java/software/wings/resources/ServiceResource.java#L88)):*
> ```Java
> @Scope(ResourceType.APPLICATION)
> @AuthRule(permissionType = PermissionType.SERVICE)
> @ApiKeyAuthorized(permissionType = PermissionType.SERVICE)
> ```

are we planning to expose all?
___
*[`400-rest/src/main/java/software/wings/resources/SettingResource.java` line 116 at r2](https://reviewable.io/reviews/harness/harness-core/36222#-N9fmju--06Ji808aYWr:-N9fmju--06Ji808aYWs:b-7ropln) ([raw file](https://github.com/harness/harness-core/blob/fe5998ed83b0bac1fa189ad8ce6d64dfc0de1fb3/400-rest/src/main/java/software/wings/resources/SettingResource.java#L116)):*
> ```Java
> @Scope(ResourceType.SETTING)
> @AuthRule(permissionType = LOGGED_IN)
> @ApiKeyAuthorized(permissionType = LOGGED_IN)
> ```

ditto
___
*[`400-rest/src/main/java/software/wings/resources/TriggerResource.java` line 65 at r2](https://reviewable.io/reviews/harness/harness-core/36222#-N9fmklNDTHTonHT-ib4:-N9fmklNDTHTonHT-ib5:b-7ropln) ([raw file](https://github.com/harness/harness-core/blob/fe5998ed83b0bac1fa189ad8ce6d64dfc0de1fb3/400-rest/src/main/java/software/wings/resources/TriggerResource.java#L65)):*
> ```Java
> @AuthRule(permissionType = LOGGED_IN)
> @ApiKeyAuthorized(permissionType = LOGGED_IN)
> public class TriggerResource {
> ```

ditto
___
*[`400-rest/src/main/java/software/wings/resources/UserGroupResource.java` line 78 at r2](https://reviewable.io/reviews/harness/harness-core/36222#-N9fmlq23D_NEfEohMcc:-N9fmlq23D_NEfEohMcd:b-7ropln) ([raw file](https://github.com/harness/harness-core/blob/fe5998ed83b0bac1fa189ad8ce6d64dfc0de1fb3/400-rest/src/main/java/software/wings/resources/UserGroupResource.java#L78)):*
> ```Java
> @Scope(ResourceType.USER)
> @AuthRule(permissionType = USER_PERMISSION_MANAGEMENT)
> @ApiKeyAuthorized(permissionType = USER_PERMISSION_MANAGEMENT)
> ```

ditto

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 5 of 17 files at r1, 4 of 5 files at r2.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36222)* status: 18 file(s) to review. 1 change requests. 1 of 2 approvals obtained.

**Comment by boopesh007** [comment]
*[`400-rest/src/main/java/software/wings/resources/ServiceResource.java` line 88 at r2](https://reviewable.io/reviews/harness/harness-core/36222#-N9fmf0c-lglcT-GkgPU:-N9fum2L5rbb5Gmy9SpS:b-cri59i) ([raw file](https://github.com/harness/harness-core/blob/fe5998ed83b0bac1fa189ad8ce6d64dfc0de1fb3/400-rest/src/main/java/software/wings/resources/ServiceResource.java#L88)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

are we planning to expose all?
</blockquote></details>

@abhinav-harness yes the ticket did not mention any particular endpoint, also since the auth rule existed already at the class level, providing the same for API Key at the same level made sense to me.

**Comment by abhinav-harness** [comment]
*[`400-rest/src/main/java/software/wings/resources/ServiceResource.java` line 88 at r2](https://reviewable.io/reviews/harness/harness-core/36222#-N9fmf0c-lglcT-GkgPU:-N9gojDk7D5ZjnKQ8lw7:bjkuylw) ([raw file](https://github.com/harness/harness-core/blob/fe5998ed83b0bac1fa189ad8ce6d64dfc0de1fb3/400-rest/src/main/java/software/wings/resources/ServiceResource.java#L88)):*
<details><summary><i>Previously, boopesh007 (Boopesh Shanmugam) wrote…</i></summary><blockquote>

@abhinav-harness yes the ticket did not mention any particular endpoint, also since the auth rule existed already at the class level, providing the same for API Key at the same level made sense to me. 
</blockquote></details>

its a bit risky though but i think we should be ok.

**Review by abhinav-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36222)* status: 18 file(s) to review.

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 13 of 17 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36222)* status: :shipit: complete! ✓

File: 400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java [hunk 1]
```diff
@@ -636,6 +637,7 @@ public RestResponse<Void> deleteDelegateGroup(@PathParam("delegateGroupId") @Not
   @ExceptionMetered
   @AuthRule(permissionType = ACCOUNT_MANAGEMENT)
   @AuthRule(permissionType = MANAGE_DELEGATES)
+  @ApiKeyAuthorized(permissionType = MANAGE_DELEGATES)
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java:640]: this should have both perms

File: 400-rest/src/main/java/software/wings/resources/DelegateSetupResourceV3.java [hunk 2]
```diff
@@ -669,6 +671,7 @@ public void setExcludeScopeIds(List<String> excludeScopeIds) {
   @ExceptionMetered
   @AuthRule(permissionType = ACCOUNT_MANAGEMENT)
   @AuthRule(permissionType = MANAGE_DELEGATES)
+  @ApiKeyAuthorized(permissionType = MANAGE_DELEGATES)
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/resources/DelegateSetupResourceV3.java:674]: both perms

**Pull request merged**